pkg/vcdclient/gateway.go

```diff
@@ -794,6 +795,12 @@ func (client *Client) createVirtualService(ctx context.Context, virtualServiceNa
 			taskURL, err)
 	}
 
+	// update RDE with freeIp
```

Comment:
Can we gate this functionality at the top itself. If there is no ClusterID, can we just skip this operation? This is to ensure that people who do not use CSE to install the cluster will still be able to use CPI/CSI.